docs/releasehistory.md

```diff
@@ -17,6 +17,14 @@ Releases follow the `major.minor.micro` scheme recommended by [PEP440](https://w
   the built-in `dict`.
   `attach_units`, `detach_units`, and `extract_serialized_units_from_dict` have been removed from
   `openff.toolkit.utils.utils`.
+- [PR #1472](https://github.com/openforcefield/openff-toolkit/pull/1472):
+  Removes SMIRKS validation while loading force fields, which previously did checked that the
+  pattern was valid SMIRKS but did not check that it represented the particular valence type it was
+  meant to represent.  This also deprecates `ChemicalEnvironment` and, by extension, the
+  `openff.toolkit.typing.chemistry` submodule.
```

Comment:
(blocking) I don't think it even checked that it was a valid SMIRKS :-/  It's be better to just mention the added deprecation warning and immediate removal of `ParameterHandler._VALENCE_TYPE` (and a mention that the field was never used in the first place)

Reply:
I think it was, but I agree with your suggestion

```
In [17]: BondHandler.BondType(
    ...:     smirks="[#1:1]-[#6:2]~[#8:3]",
    ...:     k=unit.Quantity(1.0, unit.kilocalorie_per_mole / unit.nanometer**2),
    ...:     length=unit.Quantity(1.0, unit.nanometer),
    ...: )
Out[17]: <BondType with smirks: [#1:1]-[#6:2]~[#8:3]  length: 1.0 nanometer  k: 1.0 kilocalorie_per_mole / nanometer ** 2  >

In [18]: BondHandler.BondType(
    ...:     smirks="magic",
    ...:     k=unit.Quantity(1.0, unit.kilocalorie_per_mole / unit.nanometer**2),
    ...:     length=unit.Quantity(1.0, unit.nanometer),
    ...: )
Warning: Unable to parse SMARTS
Warning: magic
Warning: ^
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File ~/software/openff-toolkit/openff/toolkit/typing/engines/smirnoff/parameters.py:430, in ParameterAttribute._call_converter(self, value, instance)
    428 try:
    429     # Static function.
--> 430     return self._converter(value)
    431 except TypeError:
    432     # Instance method.

TypeError: ParameterType.smirks() missing 2 required positional arguments: 'attr' and 'smirks'

During handling of the above exception, another exception occurred:

SMIRKSParsingError                        Traceback (most recent call last)
Cell In [18], line 1
----> 1 BondHandler.BondType(
      2     smirks="magic",
      3     k=unit.Quantity(1.0, unit.kilocalorie_per_mole / unit.nanometer**2),
      4     length=unit.Quantity(1.0, unit.nanometer),
      5 )

File ~/software/openff-toolkit/openff/toolkit/typing/engines/smirnoff/parameters.py:2488, in BondHandler.BondType.__init__(self, **kwargs)
   2483     if not has_length_bondorder:
   2484         raise SMIRNOFFSpecError(
   2485             "Either length or length_bondorder* must be specified."
   2486         )
-> 2488 super().__init__(**kwargs)

File ~/software/openff-toolkit/openff/toolkit/typing/engines/smirnoff/parameters.py:1769, in ParameterType.__init__(self, smirks, allow_cosmetic_attributes, **kwargs)
   1767 # This is just to make smirks a required positional argument.
   1768 kwargs["smirks"] = smirks
-> 1769 super().__init__(allow_cosmetic_attributes=allow_cosmetic_attributes, **kwargs)

File ~/software/openff-toolkit/openff/toolkit/typing/engines/smirnoff/parameters.py:860, in _ParameterAttributeHandler.__init__(self, allow_cosmetic_attributes, **kwargs)
    858 for key, val in smirnoff_data.items():
    859     if key in allowed_attributes:
--> 860         setattr(self, key, val)
    861     # Handle all unknown kwargs as cosmetic so we can write them back out
    862     elif allow_cosmetic_attributes:

File ~/software/openff-toolkit/openff/toolkit/typing/engines/smirnoff/parameters.py:1175, in _ParameterAttributeHandler.__setattr__(self, key, value)
   1170         raise MissingIndexedAttributeError(
   1171             f"'{key}' is out of bounds for indexed attribute '{attr_name}'"
   1172         )
   1174 # Forward the request to the next class in the MRO.
-> 1175 super().__setattr__(key, value)

File ~/software/openff-toolkit/openff/toolkit/typing/engines/smirnoff/parameters.py:379, in ParameterAttribute.__set__(self, instance, value)
    377 def __set__(self, instance, value):
    378     # Convert and validate the value.
--> 379     value = self._convert_and_validate(instance, value)
    380     setattr(instance, self._name, value)

File ~/software/openff-toolkit/openff/toolkit/typing/engines/smirnoff/parameters.py:397, in ParameterAttribute._convert_and_validate(self, instance, value)
    395 value = self._validate_units(value)
    396 # Call the custom converter before setting the value.
--> 397 value = self._call_converter(value, instance)
    398 return value

File ~/software/openff-toolkit/openff/toolkit/typing/engines/smirnoff/parameters.py:433, in ParameterAttribute._call_converter(self, value, instance)
    430         return self._converter(value)
    431     except TypeError:
    432         # Instance method.
--> 433         return self._converter(instance, self, value)
    434 return value

File ~/software/openff-toolkit/openff/toolkit/typing/engines/smirnoff/parameters.py:1750, in ParameterType.smirks(self, attr, smirks)
   1742 @smirks.converter
   1743 def smirks(self, attr, smirks):
   1744     # Validate the SMIRKS string to ensure it matches the expected
   (...)
   1748     # TODO: Add check to make sure we can't make tree non-hierarchical
   1749     #       This would require parameter type knows which ParameterList it belongs to
-> 1750     ChemicalEnvironment.validate_smirks(smirks, validate_valence_type=True)
   1751     return smirks

File ~/software/openff-toolkit/openff/toolkit/typing/chemistry/environment.py:152, in ChemicalEnvironment.validate_smirks(cls, smirks, validate_parsable, validate_valence_type, toolkit_registry)
    118 @classmethod
    119 def validate_smirks(
    120     cls,
   (...)
    124     toolkit_registry=None,
    125 ):
    126     """
    127     Check the provided SMIRKS string is valid, and if requested, tags atoms appropriate to the
    128     specified valence type.
   (...)
    150         and validate_valence_type=True
    151     """
--> 152     cls(
    153         smirks,
    154         validate_parsable=validate_parsable,
    155         validate_valence_type=validate_valence_type,
    156         toolkit_registry=toolkit_registry,
    157     )

File ~/software/openff-toolkit/openff/toolkit/typing/chemistry/environment.py:84, in ChemicalEnvironment.__init__(self, smirks, label, validate_parsable, validate_valence_type, toolkit_registry)
     82 self.label = label
     83 if validate_parsable or validate_valence_type:
---> 84     self.validate(
     85         validate_valence_type=validate_valence_type,
     86         toolkit_registry=toolkit_registry,
     87     )

File ~/software/openff-toolkit/openff/toolkit/typing/chemistry/environment.py:110, in ChemicalEnvironment.validate(self, validate_valence_type, toolkit_registry)
     89 def validate(self, validate_valence_type=True, toolkit_registry=None):
     90     """
     91     Returns True if the underlying smirks is the correct valence type, False otherwise. If the expected type
     92     is None, this method always returns True.
   (...)
    108         and validate_valence_type=True
    109     """
--> 110     perceived_type = self.get_type(toolkit_registry=toolkit_registry)
    111     if validate_valence_type and self._expected_type is not None:
    112         if perceived_type != self._expected_type:

File ~/software/openff-toolkit/openff/toolkit/typing/chemistry/environment.py:189, in ChemicalEnvironment.get_type(self, toolkit_registry)
    185     unique_tags, conn = toolkit_registry.get_tagged_smarts_connectivity(
    186         self.smirks
    187     )
    188 else:
--> 189     unique_tags, conn = toolkit_registry.call(
    190         "get_tagged_smarts_connectivity", self.smirks
    191     )
    193 if unique_tags == (1,):
    194     if len(conn) == 0:

File ~/software/openff-toolkit/openff/toolkit/utils/toolkit_registry.py:356, in ToolkitRegistry.call(self, method_name, raise_exception_types, *args, **kwargs)
    354             for exception_type in raise_exception_types:
    355                 if isinstance(e, exception_type):
--> 356                     raise e
    357             errors.append((toolkit, e))
    359 # No toolkit was found to provide the requested capability
    360 # TODO: Can we help developers by providing a check for typos in expected method names?

File ~/software/openff-toolkit/openff/toolkit/utils/toolkit_registry.py:352, in ToolkitRegistry.call(self, method_name, raise_exception_types, *args, **kwargs)
    350 method = getattr(toolkit, method_name)
    351 try:
--> 352     return method(*args, **kwargs)
    353 except Exception as e:
    354     for exception_type in raise_exception_types:

File ~/software/openff-toolkit/openff/toolkit/utils/openeye_wrapper.py:2654, in OpenEyeToolkitWrapper.get_tagged_smarts_connectivity(self, smarts)
   2652 status = oechem.OEParseSmarts(qmol, smarts)
   2653 if not status:
-> 2654     raise SMIRKSParsingError(
   2655         f"OpenEye Toolkit was unable to parse SMIRKS {smarts}"
   2656     )
   2658 unique_tags = set()
   2659 connections = set()

SMIRKSParsingError: OpenEye Toolkit was unable to parse SMIRKS magic

Reply:
Holy traceback, batman!

Reply:
Haha, that's quite a traceback!

But it's important to note that the BondType constructor isn't used when loading OFFXMLs - file loading goes through a different code path that skips validation. So the following code (which puts clearly invalid smirks in an OFFXML) runs just fine:

```python
from openff.toolkit import ForceField
ff = ForceField("openff-2.0.0.offxml")
ff.to_file('temp.offxml')
data = open('temp.offxml').read()
data = data.replace("#6", "magic")
with open('temp.offxml', 'w') as of:
    of.write(data)
ff = ForceField('temp.offxml')
```

```
<BondType with smirks: [magicX4:1]-[magicX4:2]  id: b1  length: 1.52190126495 angstrom  k: 529.2429715351 kilocalorie / angstrom ** 2 / mole  >
```